src/main/java/jst/shop/ShopService.java

```diff
@@ -13,11 +14,11 @@ public ShopService(ProductRepo productRepo) {
         orderRepo = new OrderRepo();
     }
 
-    public Product getProductbyID(int id) {
+    public Optional<Product> getProductbyID(int id) {
         try {
-            return productRepo.getProductByID(id);
+            return Optional.of(productRepo.getProductByID(id));
```

Comment:
Besser ist: bereits im ProductRepo in ein Optional zu packen. Dann brauchen wir die NoSuchElementException nicht mehr.